core/src/types/ethereum.rs

```diff
@@ -0,0 +1,94 @@
+//! Types to do with Ethereum.
+
+use std::fmt;
+use std::num::NonZeroU64;
+use std::ops::{Add, AddAssign, Deref};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use num256::Uint256;
+
+/// This type must be able to represent any valid Ethereum block height. It must
+/// also be Borsh serializeable, so that it can be stored in blockchain storage.
+///
+/// In Ethereum, the type for block height is an arbitrary precision integer - see <https://github.com/ethereum/go-ethereum/blob/v1.10.26/core/types/block.go#L79>.
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct BlockHeight {
+    inner: Uint256,
```

Comment:
This `Uint256` type is what we (and our web3 crate) are mostly using to represent Ethereum block heights, we need something Borsh-serializeable for storage though